File: testdata/bench/go/BenchmarkTest00087.go
package bench

import (
	"net/http"
	"text/template"
)

// Template injection via PostFormValue
func Handler00087(w http.ResponseWriter, r *http.Request) {
	body := r.PostFormValue("body")
	t, _ := template.New("t").Parse(body)
	t.Execute(w, nil)
}

File: testdata/bench/rust/BenchmarkTest00087.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: XSS via img src attribute injection
async fn handler(form: web::Form<HashMap<String, String>>) -> HttpResponse {
    let param = form.get("avatar").unwrap_or(&String::new()).clone();
    let html = format!("<img src='{}'>", param);
    HttpResponse::Ok().content_type("text/html").body(html)
}
